Index: clang-tidy/CMakeLists.txt
===================================================================
--- clang-tidy/CMakeLists.txt	(revision 269591)
+++ clang-tidy/CMakeLists.txt	(working copy)
@@ -30,6 +30,7 @@
 add_subdirectory(llvm)
 add_subdirectory(cppcoreguidelines)
 add_subdirectory(google)
+add_subdirectory(surelogic)
 add_subdirectory(misc)
 add_subdirectory(modernize)
 add_subdirectory(performance)
Index: clang-tidy/tool/CMakeLists.txt
===================================================================
--- clang-tidy/tool/CMakeLists.txt	(revision 269591)
+++ clang-tidy/tool/CMakeLists.txt	(working copy)
@@ -13,6 +13,7 @@
   clangTidyCERTModule
   clangTidyCppCoreGuidelinesModule
   clangTidyGoogleModule
+  clangTidySureLogicModule
   clangTidyLLVMModule
   clangTidyMiscModule
   clangTidyModernizeModule
Index: clang-tidy/tool/ClangTidyMain.cpp
===================================================================
--- clang-tidy/tool/ClangTidyMain.cpp	(revision 269591)
+++ clang-tidy/tool/ClangTidyMain.cpp	(working copy)
@@ -367,6 +367,11 @@
 static int LLVM_ATTRIBUTE_UNUSED GoogleModuleAnchorDestination =
     GoogleModuleAnchorSource;
 
+// This anchor is used to force the linker to link the GoogleModule.
+extern volatile int SureLogicModuleAnchorSource;
+static int LLVM_ATTRIBUTE_UNUSED SureLogicModuleAnchorDestination =
+    SureLogicModuleAnchorSource;
+
 // This anchor is used to force the linker to link the MiscModule.
 extern volatile int MiscModuleAnchorSource;
 static int LLVM_ATTRIBUTE_UNUSED MiscModuleAnchorDestination =
